src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.h

```diff
@@ -70,14 +90,36 @@ struct CreaseDefenderFSM
         DEFINE_SML_STATE(MoveFSM)
         DEFINE_SML_EVENT(Update)
         DEFINE_SML_SUB_FSM_UPDATE_ACTION(blockThreat, MoveFSM)
+        DEFINE_SML_STATE(DribbleFSM)
+        DEFINE_SML_GUARD(ballNearbyWithoutThreat)
+        DEFINE_SML_SUB_FSM_UPDATE_ACTION(prepareGetPossession, DribbleFSM)
 
         return make_transition_table(
             // src_state + event [guard] / action = dest_state
-            *MoveFSM_S + Update_E / blockThreat_A, MoveFSM_S = X,
+            *MoveFSM_S + Update_E[ballNearbyWithoutThreat_G] / prepareGetPossession_A =
+                DribbleFSM_S,
+            MoveFSM_S + Update_E / blockThreat_A, MoveFSM_S = X,
+            DribbleFSM_S + Update_E[!ballNearbyWithoutThreat_G] / blockThreat_A =
+                MoveFSM_S,
+            X + Update_E[ballNearbyWithoutThreat_G] / prepareGetPossession_A =
+                DribbleFSM_S,
             X + Update_E / blockThreat_A = MoveFSM_S);
     }
 
    private:
+    /** Max ratio between distances (crease and ball) / (crease and nearest enemy) for
+     * crease to chase ball. Scale from (0, 1)
+     * |------------------------------------------|
+     * | Crease | <-----------------------> Enemy |
+     * |        | <----> Ball                 |   |
+     * |       ()         x                  ()   |
+     * |------------------------------------------|
+     */
+    static constexpr double MAX_GET_BALL_RATIO_THRESHOLD = 0.3;
+    // Max distance that the crease will try and get possession of a ball
+    static constexpr double MAX_GET_BALL_RADIUS_M = 1;
+    // Max speed of ball that crease will try and get possession
+    static constexpr double MAX_BALL_SPEED_TO_GET_MS = 0.5;
```

Comment:
nit: These seem more suited to be in dynamic parameters in `parameters.proto`. I can see us potentially wanting to change some of these values in game.